File: aoc2019/day09/src/main/java/de/havox_design/aoc2016/day09/Day09.java
package de.havox_design.aoc2016.day09;

import de.havox_design.aoc.utils.java.input.DataReader;

import java.util.List;

public class Day09 {
    private final List<String> input;

    public Day09(String fileName) {
        input = readData(fileName);
    }

    public static long solvePart1(String fileName) {
        Day09 instance = new Day09(fileName);
        return instance.solvePart1();
    }

    public static long solvePart2(String fileName) {
        Day09 instance = new Day09(fileName);
        return instance.solvePart2();
    }

    public long solvePart1() {
        return 0L;
    }

    public long solvePart2() {return 0L;
    }

    private List<String> readData(String fileName) {
        return DataReader.readData(fileName, MainClass.class);
    }
}

File: templates/scala/aoc20xx-scala/src/main/scala/de/havox_design/aoc20xx/day09/Day09.scala
package de.havox_design.aoc2022.day09

object Day09 {
  def solvePart1(filename: String): Long = {
    1L
  }

  def solvePart2(filename: String): Long = {
    1L
  }

  def main(args: Array[String]): Unit = {
    println("Solution for part1: " + solvePart1("de/havox_design/aoc20xx/day09/day09.txt"))
    println("Solution for part2: " + solvePart2("de/havox_design/aoc20xx/day09/day09.txt"))
  }

  private def readData(filename: String): Iterator[String] =
    scala
      .io
      .Source
      .fromResource(filename)
      .getLines()
}
